lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +26,234 @@ def load_team_membership_csv(course, response):
+        self.course = course
+        self.max_errors = 0
+        self.existing_course_team_memberships = {}
+        self.existing_course_teams = {}
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return not self.validation_errors
+
+    def set_team_membership_from_csv(self, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+        """
+        reader = csv.DictReader((line.decode('utf-8-sig').strip() for line in input_file.readlines()))
+        self.teamset_ids = reader.fieldnames[2:]
+        row_dictionaries = []
+        csv_usernames = set()
+        if not self.validate_teamsets():
+            return False
+        self.load_user_ids_by_teamset_id()
+        self.load_course_team_memberships()
+        self.load_course_teams()
+        # process student rows:
+        for row in reader:
+            username = row['user']
+            if not username:
+                continue
+            if not self.is_username_unique(username, csv_usernames):
+                return False
+            csv_usernames.add(username)
+            user = self.get_user(username)
+            if user is None:
+                continue
+            if not self.validate_user_enrolled_in_course(user):
+                row['user'] = None
+                continue
+            row['user'] = user
+
+            if not self.validate_user_assignment_to_team_and_teamset(row):
+                return False
+            row_dictionaries.append(row)
+
+        if not self.validation_errors:
+            for row in row_dictionaries:
+                self.add_user_to_team(row)
+            return True
+        else:
+            return False
+
+    def load_course_team_memberships(self):
+        """
+        Caches existing team memberships by (user_id, teamset_id)
+        """
+        for membership in CourseTeamMembership.get_memberships(course_ids=[self.course.id]):
+            user_id = membership.user_id
+            teamset_id = membership.team.topic_id
+            self.existing_course_team_memberships[(user_id, teamset_id)] = membership.team.id
+
+    def load_course_teams(self):
+        """
+        Caches existing course teams by (team_name, topic_id)
+        """
+        for team in CourseTeam.objects.filter(course_id=self.course.id):
+            self.existing_course_teams[(team.name, team.topic_id)] = team
+
+    def validate_teamsets(self):
+        """
+        Validates team set names. Returns true if there are no errors.
+        The following conditions result in errors:
+        Teamset does not exist
+        Teamset id is duplicated
+        Also populates the teamset_names_list.
+        header_row is the list of teamset_ids
+        """
+        teamset_ids = {ts.teamset_id for ts in self.course.teams_configuration.teamsets}
```

Comment:
Nit: this method body would read better if `teamset_ids` was named `valid_teamset_ids`.